python/pyspark/pandas/frame.py

```diff
@@ -3459,6 +3458,108 @@ def mask(
         cond_inversed = cond._apply_series_op(lambda psser: ~psser)
         return self.where(cond_inversed, other)
 
+    # TODO: Support axis as 1 or 'columns'
+    def mode(
+        self, axis: Union[int, str] = 0, numeric_only: bool = False, dropna: bool = True
+    ) -> "DataFrame":
+        """
+        Get the mode(s) of each element along the selected axis.
+
+        The mode of a set of values is the value that appears most often.
+        It can be multiple values.
+
+        .. note:: the current implementation of mode requires joins
+            multiple times(columns count - 1 times when axis is 0 or 'index'),
+            which is potentially expensive.
+
+        .. note:: the order of multiple modes (within a column when axis is 0 or 'index')
+            is not determined.
+
+        Parameters
+        ----------
+        axis : {0 or 'index', 1 or 'columns'}, default 0
+            The axis to iterate over while searching for the mode:
+            * 0 or 'index' : get mode of each column
+            * 1 or 'columns' : get mode of each row.
+
+        numeric_only : bool, default False
+            If True, only apply to numeric columns.
+
+        dropna : bool, default True
+            Don't consider counts of NaN/NaT.
+
+        Returns
+        -------
+        DataFrame
+            The modes of each column or row.
+
+        See Also
+        --------
+        Series.mode : Return the highest frequency value in a Series.
+        Series.value_counts : Return the counts of values in a Series.
+
+        Examples
+        --------
+        >>> psdf = ps.DataFrame(
+        ...     [("bird", 2, 2), ("mammal", 4, np.nan), ("arthropod", 8, 0), ("bird", 2, np.nan)],
+        ...     index=("falcon", "horse", "spider", "ostrich"),
+        ...     columns=("species", "legs", "wings"),
+        ... )
+        >>> psdf
+                   species  legs  wings
+        falcon        bird     2    2.0
+        horse       mammal     4    NaN
+        spider   arthropod     8    0.0
+        ostrich       bird     2    NaN
+        >>> psdf.mode()  # doctest: +SKIP
```

Comment:
Skip the doctest because the order of multiple modes within a column is not determined.

Reply:
How about use `sort_index()` ??

Reply:
How about `psdf.mode().sort_index(axis=1)` ??

Reply:
Unfortunately, for each column, the order of multiple modes within the column is undetermined. I am not sure `sort_index` works in this case.

Reply:
I added `each column` to `Notes` for clarification.